corehq/apps/hqadmin/management/commands/clone_domain.py

```diff
@@ -0,0 +1,211 @@
+from optparse import make_option
+
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.userreports.dbaccessors import get_report_configs_for_domain, get_datasources_for_domain
+
+
+class Command(BaseCommand):
+    args = "<existing_domain> <new_domain>"
+    help = """Clone a domain and it's data (settings, fixtures, locations, products, UCR, apps)"""
+
+    option_list = BaseCommand.option_list + (
+        make_option('--no-flags', action='store_false', dest='flags', default=True,
+                    help="Don't process domain flags"),
+        make_option('--no-fixtures', action='store_false', dest='fixtures', default=True,
+                    help="Don't process fixtures"),
+        make_option('--no-locations', action='store_false', dest='locations', default=True,
+                    help="Don't process locations"),
+        make_option('--no-products', action='store_false', dest='products', default=True,
+                    help="Don't process products"),
+        make_option('--no-ucr-or-apps', action='store_false', dest='ucr_apps', default=True,
+                    help="Don't process UCR"),
+    )
+
+    def handle(self, *args, **options):
+        existing_domain, new_domain = args
+        self.clone_domain_and_settings(existing_domain, new_domain)
+        if options['flags']:
+            self.set_flags(existing_domain, new_domain)
+
+        if options['fixtures']:
+            self.copy_fixtures(existing_domain, new_domain)
+
+        if options['locations']:
+            self.copy_locations(existing_domain, new_domain)
+
+        if options['products']:
+            self.copy_products(existing_domain, new_domain)
+
+        if options['ucr_apps']:
+            report_map = self.copy_ucr_data(existing_domain, new_domain)
+            self.copy_applications(existing_domain, new_domain, report_map)
+
+    def clone_domain_and_settings(self, old_domain, new_domain):
+        from corehq.apps.domain.models import Domain
+        new_domain_obj = Domain.get_by_name(new_domain)
+        if new_domain_obj:
+            if raw_input(
+                '{} domain already exists. Do you still want to continue? [y/n]'.format(new_domain)
+                ).lower() == 'y':
```

Comment:
E125 continuation line with same indent as next logical line
E123 closing bracket does not match indentation of opening bracket's line